console/src/main/scripts/plugins/metrics/ts/app-details/appServerDatasourcesDetails.ts

```diff
@@ -568,16 +473,75 @@ module HawkularMetrics {
 
     }
 
+    public redirectToDataSource(resource, event) {
+      if (this.canRedirect(event.target)) {
+        let timeOffset = (this.$routeParams.timeOffset) ? this.$routeParams.timeOffset : 0;
+        let endTime = (this.$routeParams.endTime) ? this.$routeParams.endTime : 0;
+        let redirectUrl = AppServerDatasourcesDetailsController.BASE_URL +
+          '/' + this.$routeParams.feedId +
+          '/' + this.$routeParams.resourceId +
+          '/' + this.$routeParams.tabId +
+          '/' + this.encodeResourceId(resource.id) +
+          '/' + timeOffset +
+          '/' + endTime;
+        this.$location.path(redirectUrl);
+      }
+    }
+
+    private canRedirect(clickedElement):boolean {
+      let tags = ['button', 'a'];
+      return !(
+        tags.indexOf(clickedElement.tagName.toLowerCase()) !== -1 ||
+        clickedElement.classList.contains('caret')
+      );
+    }
+
     public encodeResourceId(resourceId:ResourceId):string {
```

Comment:
This is not the only place that this kind of thing occurs. Lets consolidate these:
https://github.com/hawkular/hawkular/blob/0421ffed81484d92cfe2ee96ad529343a4b6c786/console/src/main/scripts/plugins/metrics/ts/utility.ts#L25-L25